src/query.rs

```diff
@@ -18,4 +18,16 @@ impl Query {
     fn __repr__(&self) -> PyResult<String> {
         Ok(format!("Query({:?})", self.get()))
     }
+
+    pub fn __copy__(&self) -> Self {
+        Self {
+            inner: self.get().box_clone(),
```

Comment:
Maybe implement the usual `Clone` trait manually and then use the macros? (But again, this class seems immutable and `#[pyclass(frozen)]` appears preferable to copy support.)